web-app/src/components/BookingCalendar/container.js

```diff
@@ -41,52 +46,35 @@ const BookingCalendarContainer = Wrapped =>
       this.props.setEventBeingUpdated(isBeingUpdated);
     };
 
-    checkIfPastDatesSelected = start => {
-      const today = new moment();
-      return moment(start).isBefore(today);
-    };
-
-    selectedDatesOverlapExisting = (start, end) => {
+    handleCalendarValidation = (start, end) => {
       const { events, employeeId } = this.props;
-      const overlappingEvents = events.filter(event => {
-        const { employee } = event;
-        if (employee && employee.employeeId === employeeId) {
-          var selectedDateRange = moment.range(
-            moment(start),
-            moment(end).endOf('day')
-          );
-          var existingEvent = moment.range(
-            moment(event.start),
-            moment(event.end)
-          );
-          if (selectedDateRange.overlaps(existingEvent)) {
-            return true;
-          }
-        }
-      });
-      return overlappingEvents.length > 0;
-    };
-
-    validateSelectedDates = (start, end) => {
-      const pastDatesSelected = this.checkIfPastDatesSelected(start);
+      const pastDatesSelected = checkIfPastDatesSelected(start);
+      const datesFallOnWeekend = checkIfDatesFallOnWeekend(start, end);
       if (pastDatesSelected) {
         return 'Unable to select past dates';
+      } else if (datesFallOnWeekend) {
+        return 'Unable to select weekend dates';
       } else {
-        const datesOverlapExisting = this.selectedDatesOverlapExisting(
+        const datesOverlapExisting = checkIfSelectedDatesOverlapExisting(
+          events,
+          employeeId,
           start,
-          end
+          end,
         );
         if (datesOverlapExisting) {
-          return 'You\'re are trying to request dates that have already been set';
+          return 'You cannot request dates that have already been set';
         } else {
           return 'Dates approved';
         }
       }
     };
 
     onSelectSlot = ({ start, end }) => {
-      const validatingDatesResult = this.validateSelectedDates(start, end);
-      if (validatingDatesResult === 'Dates approved') {
+      const calendarValidationResults = this.handleCalendarValidation(
+        start,
+        end,
+      );
+      if (calendarValidationResults === 'Dates approved') {
```

Comment:
I'm still not a huge fan of this error/success message string being used in part of the logic. It's used in quite a lot of places here too.

A lot would break if we wanted to display any other success message to the user.

Reply:
Some things are totally obvious once pointed out. We've used Const files throughout the application Gary which works well. Let me tidy that up.